packages/cli-e2e/utils/login.ts

```diff
@@ -0,0 +1,121 @@
+import {deletePassword, getPassword} from 'keytar';
+import {userInfo} from 'os';
+import {Page, Target} from 'puppeteer-core';
+import type {Browser} from 'puppeteer-core';
+import {ChildProcessWithoutNullStreams, spawn} from 'child_process';
+import {answerPrompt, CLI_EXEC_PATH, isYesNoPrompt} from './cli';
+
+function isLoginPage(page: Page) {
+  // TODO: CDX-98: URL should vary in fonction of the targeted environment.
+  return page.url().match(/https:\/\/platform.*cloud\.coveo\.com\/login/);
+}
+
+export async function isLoggedin() {
+  const currentAccount = userInfo().username;
+  const accessToken = await getPassword(
+    'com.coveo.cli.access.token',
+    currentAccount
+  );
+  return accessToken !== null;
+}
+
+function waitForLoginPage(browser: Browser) {
+  return new Promise<Page>((resolve) => {
+    browser.on('targetchanged', async (target: Target) => {
+      const page = await target.page();
+      if (page && isLoginPage(page) !== null) {
+        resolve(page);
+      }
+    });
+  });
+}
+
+async function staySignedIn(page: Page) {
+  await page.waitForSelector('input[type="submit"]');
+  await page.click('input[type="submit"]');
+}
+
+async function possiblyAcceptCustomerAgreement(page: Page) {
+  if (page.url().match(/https:\/\/platform.*cloud\.coveo\.com\/eula.?/)) {
```

Comment:
Mhhh. I would create a URL object from `page.url()` and manipulate it instead. (I personnally, tho it's taste/flavoring don't like using regexp for URL validations).

That's my opinion tho!

Reply:
I'll just hardcode the dev environment for now. We can update the validation in CDX-98

Reply:
add a todo comment then ;)

Reply:
Yes, I added it in my [last commit](https://github.com/coveo/cli/pull/68/commits/21e110246c15224f256806d69ddafe55ef907605#diff-f882e2d46f5ef618509857e8a5a62df6468201ee6ed6fa39b35c83a2b2d4d765R10).